op-challenger/game/keccak/matrix/matrix_test.go

```diff
@@ -220,6 +195,126 @@ func TestMatrix_AbsorbNextLeaf(t *testing.T) {
 	}
 }
 
+func TestVerifyPreimage(t *testing.T) {
+	preimage := testutils.RandomData(rand.New(rand.NewSource(2323)), 1024)
+	validCommitments := func() []common.Hash {
+		s := NewStateMatrix()
+		valid, err := s.AbsorbUpTo(bytes.NewReader(preimage), 1000*types.BlockSize)
+		require.ErrorIs(t, err, io.EOF, "Should read all preimage data")
+		return valid.Commitments
+	}
+	leafData := func(idx int) [types.BlockSize]byte {
+		var out [types.BlockSize]byte
+		copy(out[:], preimage[idx*types.BlockSize:(idx+1)*types.BlockSize])
+		return out
+	}
```

Comment:
Could use a named return for brevity I believe
```suggestion
	leafData := func(idx int) (out [types.BlockSize]byte) {
		copy(out[:], preimage[idx*types.BlockSize:(idx+1)*types.BlockSize])
		return
	}
```